tools/monorepo-utils/src/core/git.ts

```diff
@@ -110,6 +115,29 @@ export const cloneRepoShallow = async ( repoPath: string ) => {
 	return await cloneRepo( repoPath, { '--depth': 1 } );
 };
 
+/**
+ * Clone a repo using the authenticated token `GITHUB_TOKEN`. This allows the script to push branches to origin.
+ *
+ * @param {Object}  options       CLI options
+ * @param {string}  options.owner repo owner
+ * @param {string}  options.name  repo name
+ * @param {boolean} isShallow     whether to do a shallow clone or not.
+ * @return {string} temporary repo path
+ */
+export const cloneAuthenticatedRepo = async (
+	options: { owner: string; name: string },
+	isShallow = true
+): Promise< string > => {
+	const { owner, name } = options;
+	const source = `github.com/${ owner }/${ name }`;
+	const token = getEnvVar( 'GITHUB_TOKEN' );
+	const remote = `https://${ owner }:${ token }@${ source }`;
```

Comment:
@ObliviousHarmony I'm wondering if this may be a security risk. When you test this, the location of the temporary repo will be logged to the console. Check the remote, it will have the token in the url

```
cd <temp-repo-path>
git remote -v
```
Having that in a temporary local folder isn't great, but not too bad either. I'm more wondering if there are implications when running on actions in github. The authentication is required to push branches to Github.

Reply:
That _is_ an interesting concern. Is there some way we can do this without exposing the token?

Reply:
@psealock untested but maybe if we disable debug logging it will keep this from being a problem?

`git config --global core.debuglog false`

Reply:
dasherized in https://github.com/woocommerce/woocommerce/pull/38082/commits/65de8a7901b03ceeac0c289e60eb51bba8817cd5

Reply:
Thanks @samueljseay, although that doesn't seem to work. We aren't logging this in the script, GH bot tokens are regenerated per action, and GH prevents logging this value anyways. So I think we are ok here.